### PR TITLE
[amp-story-player] Throw error when trying to call load() on player that isn't attached to the DOM

### PR DESCRIPTION
**Pull request by Enriqe**
There will be some cases where publishers will create the player element dynamically and it will start not appended to the DOM. In these cases, when the iframe is not appended to the DOM, it will not have a `iframeEl.contentWindow`, which the messaging library uses to establish the communication.

With this PR, we will throw an error when trying to call `load()` before the player is appended to the DOM.

### CODE REVIEW

**Comment by gmajoulet** [comment]
Can you explain a bit more about this fix? The iframe will fire a `onload` when (1) connected to the DOM before you set the `src`, and (2) another one once the entire content is loaded?

Doesn't the current `onload` handler currently try to know when (2) happens, and not (1)? Is it broken?

**Comment by Enriqe** [comment]
`iframe.onload` will only fire once, when the content is loaded, and the content will only load when the iframe is appended to the DOM (not two different times). 

The iframe won't load unless it's appended to the DOM.

**Comment by gmajoulet** [comment]
`iframe.onload` can fire twice, I wrote a codepen that replicate the Player code behavior (append the iframe to the DOM, do some stuff, and then set the src) here: https://codepen.io/gmajoulet/pen/poNjVYm

- If you want to figure out when the `iframe` is connected to the DOM, this is a bit hacky and we should find something else.
- If you want to figure when the iframe is totally loaded before setting the messaging and playing the story, this will be a big performance hit and we should definitely not do it, cf MDN:

> The load event fires at the end of the document loading process. At this point, all of the objects in the document are in the DOM, and all the images, scripts, links and sub-frames have finished loading.

As a side note, I see that we use `onload = () => {}` to run code like `this.element_.classList.add(LoadStateClass.LOADED);`, I suspect we might have bugs with this code given the current implementation (onload might trigger when we append the iframe to the DOM and not when the content is loaded) and should investigate.

**Comment by Enriqe** [comment]
Documenting what was discussed offline:

The issue with the `onload` event dispatching multiple times can be solved by attaching the `onload` event to the iframe after it's appended to the DOM or by always setting the `src` on the iframe before appending it to the DOM [see codepen](https://codepen.io/maenrique/pen/poNgJMb). Also by checking the [event.target.src](https://stackoverflow.com/questions/10781880/dynamically-created-iframe-triggers-onload-event-twice/38459639#38459639) on the iframe `onload` event. But this is a separate issue that will be described and addressed in #32436.

For now, this PR will throw an error when `load()` is called before the player element is attached to the DOM, which causes the original bug that this is solving:
```
        const playerContainer = document.getElementById("story-player");
        const playerEl = document.createElement("amp-story-player");

        const player = new AmpStoryPlayer(window, playerEl);


        if (playerEl.isReady) {
            playerEl.add(stories);
        } else {
          playerEl.addEventListener('ready', () => {
            playerEl.add(stories);
          });
        }

        player.load(); // error
        playerContainer.append(playerEl);
 ```

File: src/amp-story-player/amp-story-player-impl.js [hunk 1]
```diff
@@ -261,6 +261,11 @@ export class AmpStoryPlayer {
    * @public
    */
   load() {
+    if (!this.doc_.querySelector(TAG)) {
```
Comment (gmajoulet) [inline src/amp-story-player/amp-story-player-impl.js:264]: Please use `isConnected`, there might be multiple players in the DOM

File: src/amp-story-player/amp-story-player-impl.js [hunk 2]
```diff
@@ -261,6 +261,11 @@ export class AmpStoryPlayer {
...
+      throw new Error(
+        `[${TAG}] element must be appended to the DOM before calling load().`
```
Comment (gmajoulet) [inline src/amp-story-player/amp-story-player-impl.js:266]: Nit: let's use the word `connected` that's more inline with the HTML spec

**Review by gmajoulet — COMMENTED** [review]
Cool!! Can you add one sentence in the documentation?

**Review by gmajoulet — APPROVED** [review]

**Pull request merged**